### DISCUSSION TITLE
L11_flow_direction error

### DISCUSSION THREAD

**Discussion by ViniciusBMoreira** [mhm-ufz/mHM#19] [Q&A] [answered]
Hey guys!

I am encountering this error `Error L11_flow_direction: side = -1` and I cannot identify the source. can you help me? Informations below

_Maybe it's because my dem has some flat areas, so my flow direction has too..._

[mhm.nml.txt](https://github.com/mhm-ufz/mhm/files/6652769/mhm.nml.txt)
[fdir.asc.txt](https://github.com/mhm-ufz/mhm/files/6652774/fdir.asc.txt)

```
----------------------------------------------------------------------
              mHM-UFZ

    MULTISCALE HYDROLOGIC MODEL
           Version: 5.11.2-dev0
           Date:    2021-06-14

Originally by L. Samaniego & R. Kumar
----------------------------------------------------------------------

Openmp not used.
MPI not used.
Start at 15.06.2021 01:01:10.
Working directory: /home/vbryansm/def1
Using main file mhm_driver.f90 and namelists:
     mhm.nml
     mhm_parameter.nml
     mhm_outputs.nml


Read namelist file: mhm.nml
Read namelist file: mhm_parameter.nml
Read namelist file: mhm_outputs.nml
 INFO: Process (10, neutrons) is deactivated, so output will be suppressed.

domain 1:
resolution Hydrology (domain 1)     = 10000.
resolution Routing (domain 1)       = 10000.

Resolution of routing and hydrological modeling are equal!

Following output will be written:
  NetCDF deflate level: 9
  NetCDF output precision: single
  STATES:
    interceptional storage                          (L1_inter) [mm]
    soil water content in the single layers     (L1_soilMoist) [mm]
    volumetric soil moisture in the single layers              [mm/mm]
    mean volum. soil moisture averaged over all soil layers    [mm/mm]
    waterdepth in reservoir of sealed areas       (L1_sealSTW) [mm]
    waterdepth in reservoir of unsat. soil zone  (L1_unsatSTW) [mm]
    waterdepth in reservoir of sat. soil zone      (L1_satSTW) [mm]
  FLUXES:
    potential evapotranspiration PET                  (L1_pet) [mm/T]
    actual evapotranspiration aET               (L1_aETCanopy) [mm/T]
    total discharge generated per cell       (L1_total_runoff) [mm/T]
    direct runoff generated per cell           (L1_runoffSeal) [mm/T]
    fast interflow generated per cell          (L1_fastRunoff) [mm/T]
    slow interflow generated per cell          (L1_slowRunoff) [mm/T]
    baseflow generated per cell                  (L1_baseflow) [mm/T]
    groundwater recharge                           (L1_percol) [mm/T]
    infiltration                                (L1_infilSoil) [mm/T]
    actual evapotranspiration from soil layers    (L1_aETSoil) [mm/T]
    effective precipitation                     (L1_preEffect) [mm/T]

FINISHED reading config

  Inititalize mRM

    Following output will be written:
    NetCDF deflate level: 9
    NetCDF output precision: single
    FLUXES:
      routed streamflow      (L11_qMod)                [m3 s-1]

    FINISHED reading config


# of domains:                  1

  Input data directories:
  --------------
      DOMAIN                    1
  --------------
    Morphological directory:      test_domain/input/morph/ (found)
    Land cover directory:         test_domain/input/luse/ (found)
    Precipitation directory:      test_domain/input/meteo/pre/ (found)
    Temperature directory:        test_domain/input/meteo/tavg/ (found)
    PET directory:                test_domain/input/meteo/pet/ (found)
    Output directory:             test_domain/output_b1/ (found)


  Read data ...
  Reading data ...
    Reading dem and lcover ...
      Reading dem for domain: 1 ...
      Reading lcover for domain: 1 ...

Tillage layers: the tillage horizons are modelled
                upto mHM layers:          1
    Reading data for domain: 1 ...
      Reading slope ...
      Reading aspect ...
      Reading soil ids ...
      Reading LAI ...
      Reading latitude/logitude ...
    in     0.307 seconds.
  in     0.405 seconds.
***WARNING: Geological unit 2
            is not appearing in study domain.
***WARNING: Geological unit 6
            is not appearing in study domain.
***WARNING: Geological unit 8
            is not appearing in study domain.
***WARNING: Geological unit -9999
            is not appearing in study domain.
    in     0.405 seconds.
  Initialize domains ...
  in     0.330 seconds.
  Read forcing and optional data ...
    read meteo weights for tavg     ...
    read meteo weights for pet     ...
    read meteo weights for pre     ...
  Reading meteorological forcings for Domain: 1 ...
    read precipitation        ...
    read temperature          ...
    read pet                  ...
    in     3.921 seconds.
    in     3.519 seconds.
      Reading data for domain: 1 ...
STOP Error L11_flow_direction: side = -1
```

EDIT:  when I try to run with l0 = l1 = 500m, the mhm gets stuck on the screen below which apparently is part of the L11_flow_direction subroutine.

```
 Initialize domains ...
  in     0.484 seconds.
  Read forcing and optional data ...
    read meteo weights for tavg     ...
    read meteo weights for pet     ...
    read meteo weights for pre     ...
  Reading meteorological forcings for Domain: 1 ...
    read precipitation        ...
    read temperature          ...
    read pet                  ...
    in   127.984 seconds.
    in   128.850 seconds.
      Reading data for domain: 1 ...

    Domain:   1
      Number of outlets found at Level 0:..       0
      Number of outlets found at Level 11:.       1
_
```

**Comment by pallavhydro** [comment, marked as answer]
Hi Bryan

**First case**:

The `L11_flow_direction: side = -1` means one (or more) of your model routing cell (L11) is not being able to find a direction to flow. Basically, one (or more) of the L0 cells (dem.asc) is a sink. Please fill the dem you are using and update the facc.asc, fdir.asc and idgauges.asc from the filled dem. 

**Second case**:

My guess is your L1/ L11 of 500 m is too fine a resolution to run. This is making mhm take time for generating the stream network configurations. I would suggest you to stick to 10,000 m that you were using until you are confident about your setup. Once that is done, then try coming down to few kms resolution. BTW, the `m` in mhm stands for `mesoscale`. It's not made to run at fine resolutions, not because of computational reasons but the processes representation incorporated represent mesoscale processes rather than fine scale. May be @StephanThober and @lese66 can add more on this point.

Hope you get through this issue.

Pallav

**Reply by ViniciusBMoreira** [reply]
Perfect, thank you very much. I will be back in a few hours with an answer!

**Reply by ViniciusBMoreira** [reply]
@pallavhydro 

My dem is definitely filled, I tried using these nomenclatures in flow direction: `1-2-4-8-16-32-64-128`  according to this:

![image](https://user-images.githubusercontent.com/61054098/122096857-6162c580-cde5-11eb-9fba-006c875dbef4.png)

and `1-2-3-4-5-6-7-8` too.

But nothing works.

my l0 = 500m
l1/l22/l2 = 10000m.

Can you give me some explanation of how the model will generate the flow direction for l11? does it just use dem.asc?


I will redo all the data involving the dem, using another source of information

edit.:

I remade all my .asc files with another DEM source and and the program worked. But I have to restart the computer after updating the files

**Comment by lese66** [comment]
Dear Brian,

The upscale of the routing is explained at 

Thober, S., Cuntz, M., Kelbling, M., Kumar, R., Mai, J. & Samaniego, L. The multiscale routing model mRM v1.0: simple river routing at resolutions from 1 to 50 km. Geoscientific Model Development 12, 2501–2521 (2019).

In short: the coarse scale direction follows the largest accumulation ate the boundaries of the coarse cell accounted with the L0 flow direction. at L11 only N, E, W S directions are allowed.

Please analyze the test basins. Prepare a simple example until you understand the logic of the model.

Luis






> On 15. June 2021, at 19:39, Bryan Souza ***@***.***> wrote:
> 
> 
> @pallavhydro
> 
> My dem is definitely filled, I tried using these nomenclatures in flow direction: 1-2-4-8-16-32-64-128 and 1-2-3-4-5-6-7-8. according to this:
> 
> 
> But nothing works.
> 
> my l0 = 500m
> l1/l22/l2 = 10000m.
> 
> Can you give me some explanation of how the model will generate the flow direction for l11? does it just use dem.asc?
> 
> I will redo all the data involving the dem, using another source of information
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

--
Dr. Luis Samaniego / Deputy Head
Department Computational Hydrosystems (CHS)

Helmholtz-Zentrum für Umweltforschung GmbH - UFZ
Helmholtz Centre for Environmental Research - UFZ
Permoserstraße 15 / 04318 Leipzig / Germany
phone +49 341 235 1971 / fax +49 341 235 45 1971
***@***.*** / www.ufz.de

Helmholtz-Zentrum für Umweltforschung GmbH - UFZ
Registered Office: Leipzig
Registration Office: Amtsgericht Leipzig
Trade Register Nr. B 4703
Chairman of the Supervisory Board: MinDirig'in Oda Keppler
Scientific Director: Prof. Dr. Georg Teutsch
Administrative Director: Dr. Sabine König

-----------------------------------------------------------------
Stochastic and Land Surface Hydrology Group
www.ufz.de/index.php?en=13975
mHM v5.10 released www.ufz.de/mhm
www.ufz.de/droughtmonitor
www.ufz.de/ulysses
-----------------------------------------------------------------

**Reply by ViniciusBMoreira** [reply]
Hey @lese66 

Thanks for the answer.

I managed to run the model with my own basin, luckily! now i'm finding some holes in the output. But I'll go over my inputs/parameters/calibration and see what I can do about it

![image](https://user-images.githubusercontent.com/61054098/122137612-6cd0e380-ce1b-11eb-86c7-8cac97c2839e.png)


Bryan

**Reply by ViniciusBMoreira** [reply]
Hey @lese66 !

Can you take a look in https://github.com/mhm-ufz/mhm/discussions/21 ?

I think you can give me some tips !

Ty
Bryan

**Reply by rohinikumar82** [reply]
Hi @BryanSouzaM,
For the sake of better archiving of discussion on this issue, lets stick to a single discussion channel. 
I provided some points/helps in your other discussion channel... 
Hope to solve this problem soon. 
thanks. 
Best,
Rohini

**Comment by LuvMangorange** [comment]
Hello
I cannot find the standard of the hydrogeological classification in my sample besides the bgr website does not provide the hydrogeological map of China.
So can you help me?

**Reply by rohinikumar82** [reply]
Please check our response to "What`s the classification standard of the hydrogeological data #22"
